### ISSUE TITLE
Instructor view

### ISSUE THREAD

**Issue by ashwinvis** [coderefinery/sphinx-lesson#108]
One good thing about the carpentries workbench generator is that you get an instructor view, where directives such as **instructor**  and **break** are obscured from the learner view. 

This can be nice to have. So the questions which remain are:

- do we want this?
- how does one implement this in sphinx

## Example

One selects it from the top-right corner

<img src="https://github.com/user-attachments/assets/b8e89838-9732-45b8-b814-0438d4cc81c4" width=400>

### Comparison 
| Learner view | Instructor view |
|--------|--------|
| [Link](https://carpentries-incubator.github.io/deep-learning-intro/2-keras.html) | [Link](https://carpentries-incubator.github.io/deep-learning-intro/instructor/2-keras.html) |
| ![image](https://github.com/user-attachments/assets/2dacdc0f-5d85-49e4-b58c-6978e43401ea) | ![image](https://github.com/user-attachments/assets/9242c6a2-4a3d-418b-bf14-5ba386ed07f5) |

**Comment by ashwinvis** [comment]
It might be possible to add a custom HTML builder which generates the instructor view.

**Comment by samumantha** [comment]
This is indeed nice. 

We have the instructor guide as separate page or sometimes as visible boxes in the materials, I guess there was some decision as to why that is so earlier in the development?

**Comment by bast** [comment]
There is an advantage of having the instructor note rendered in the page of the material. When preparing the lesson, instructors are less likely to overlook it and more likely to keep it up-to-date. The separate pages have suffered from being overlooked and outdated.

**Comment by ashwinvis** [comment]
Another advantage is that inline tips easily show the context.

**Comment by ashwinvis** [comment]
I thought of something... in RTD you can have latest and stable versions side by side. If we call them learner and instructor instead it will work.

We might be able to acheive it without RTD using something like:

https://sphinx-versions.readthedocs.io/en/latest/


Then add a little CSS to toggle the instructor admonition between the 2 "versions"

**Comment by bast** [comment]
TIL about sphinx-versions. Thank you!

However, sphinx-versions distinguishes based on branches/tags but we probably want both versions to appear in the same branch and same source code?

**Comment by samumantha** [comment]
To keep things up to date it may be good to have it all in one branch (and maybe even file?) 👍

I wonder how people would use it:

- as an instructor when preparing for a lesson, it would be good to have it in same text?

- as a learner you do not want to be distracted by it so it should not be visible (at all)?

- as an instructor during teaching you sometimes show the materials page, then it should not be visible to learners to not distract. But especially as a new instructor reminders on things while going through them in class might be good 🤔

If we could have both learner view and instructor guide open in two different windows or the same but separate and scrolling somehow linked, could that help? Ie instructor could have the instructor stuff in not shared part of the screen?

Or might collapsible blocks be a solution (like solutions in exercises?)?

**Comment by bast** [comment]
I imagined either collapsible blocks or a button/switch which can turn visibility on/off using JavaScript.

**Comment by samumantha** [comment]
If I see it right in Ashwins original suggestion the right image also has the instructor notes as collapsible blocks? 

Unless someone has time to develop some javascript or other solution to having the button/switch between learner and instructor view, should we start with integrating the instructor notes into the lesson material as collapsible blocks? It is already possible with the current setup, isn't it?

Then we could still implement the switch later?

**Comment by bast** [comment]
Good plan. My recommendation is that instructor notes are tracked in the same file but then rendered somehow. And collapsible is a good first goal. And the nice thing is that such effort would not be wasted since one can then add a switch or differential rendering later without having to change content.

**Comment by ashwinvis** [comment]
CSS + JavaScript toggle button solution seems doable. I will think about it.
